rskj-core/src/test/java/co/rsk/peg/FederationSupportTest.java

```diff
@@ -176,71 +168,32 @@ void getFederatorPublicKeys() {
             new FederationMember(btcKey0, rskKey0, mstKey0),
             new FederationMember(btcKey1, rskKey1, mstKey1)
         );
-        FederationArgs federationArgs = new FederationArgs(members, Instant.ofEpochMilli(123), 456,
-            NetworkParameters.fromID(NetworkParameters.ID_REGTEST));
+        FederationArgs federationArgs = new FederationArgs(
+            members,
+            Instant.ofEpochMilli(123),
+            456,
+            federationMainnetConstants.getBtcParams()
+        );
         Federation theFederation = FederationFactory.buildStandardMultiSigFederation(
             federationArgs
         );
-        when(provider.getNewFederation()).thenReturn(theFederation);
-
-        Assertions.assertTrue(Arrays.equals(federationSupport.getFederatorPublicKeyOfType(0, FederationMember.KeyType.BTC), btcKey0.getPubKey()));
-        Assertions.assertTrue(Arrays.equals(federationSupport.getFederatorPublicKeyOfType(1, FederationMember.KeyType.BTC), btcKey1.getPubKey()));
+        when(storageProvider.getNewFederation(federationMainnetConstants, activations)).thenReturn(theFederation);
 
-        Assertions.assertTrue(Arrays.equals(federationSupport.getFederatorBtcPublicKey(0), btcKey0.getPubKey()));
-        Assertions.assertTrue(Arrays.equals(federationSupport.getFederatorBtcPublicKey(1), btcKey1.getPubKey()));
+        assertArrayEquals(federationSupport.getActiveFederatorPublicKeyOfType(0, KeyType.BTC), btcKey0.getPubKey());
+        assertArrayEquals(federationSupport.getActiveFederatorPublicKeyOfType(1, KeyType.BTC), btcKey1.getPubKey());
 
-        Assertions.assertTrue(Arrays.equals(federationSupport.getFederatorPublicKeyOfType(0, FederationMember.KeyType.RSK), rskKey0.getPubKey(true)));
-        Assertions.assertTrue(Arrays.equals(federationSupport.getFederatorPublicKeyOfType(1, FederationMember.KeyType.RSK), rskKey1.getPubKey(true)));
+        assertArrayEquals(federationSupport.getActiveFederatorBtcPublicKey(0), btcKey0.getPubKey());
+        assertArrayEquals(federationSupport.getActiveFederatorBtcPublicKey(1), btcKey1.getPubKey());
```

Comment:
super nit, but i would revert the order. So we have the checks for BTC, RSK and MST key one after the other

Reply:
Not sure I follow, that's how it is now